### PR TITLE
Feature/voice rec/main

### LINKED ISSUE #54: Implementing voice recieve.
### Link

https://github.com/Rapptz/discord.py/issues/444

### PR DESCRIPTION
**Pull request by veni-vidi-code**
## Summary

Implementation of basic voice recording
Closes #54 

This is a work in progress based on https://github.com/Rapptz/discord.py/pull/6507 but also working with multiple speakers

TODO
- [x] Document this code
- [x] Rework the example
- [x] Test it with more than 2 users
- [x] Stress test it over long recordings
- [x] Find out why muting does not produce a gap
- [x] Work on filters
- [x] Document filters
- [x] Test all Encoding types

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [x] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [x] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Review by Sheepposu — CHANGES_REQUESTED** [review]

**Review by VincentRPS — APPROVED** [review]
lgtm

**Review by VincentRPS — CHANGES_REQUESTED** [review]
just a suggestion ext.voicerecording imo is too long maybe you can name it ext.audio or ext.rec

**Review by RPSMain — APPROVED** [review]

File: nextcord/errors.py [hunk 1]
```diff
@@ -258,6 +263,33 @@ def __init__(self, shard_id: Optional[int]):
...
+    pass
+
+
+class NotConnected(ListeningException):
+    """Exception that's thrown when trying to listen while not connected"""
+    pass
+
+
+class AlreadyListening(ListeningException):
+    """Exception that's thrown when trying to listen while already listening"""
```
[the remark this answers is not in the document]
Comment (ooliver1) [reply nextcord/errors.py:279]: If you do this change, make sure it's not over 75 lines or whatever it is for docstrings

**Review by TAG-Epic — CHANGES_REQUESTED** [review]
Honestly I dislike the current voice receive api and we should definitely not store files to the users system without them doing it..

**Comment by ooliver1** [comment]
im gonna close this for now as its been stale for a while, feel free to reopen this once progress resumes

**Comment by Mistobaan** [comment]
What happened to this PR? @ooliver1 stale dev is the only issue?

**Comment by laithalshabi** [comment]
how to use it where is the documentation <3

**Pull request closed without merging**